File: 13_NoSQL/Uebungsdb Exams/Model/Exam.cs
﻿using MongoDB.Bson;
using System;

namespace ExamDbGenerator.Model
{
    record Exam(StudentName Student, TeacherName Teacher, Class ExamClass, Subject Subject, DateTime DateTime, int PointsMax, int Points, int Grade)
    {
        public ObjectId Id { get; private set; } = ObjectId.GenerateNewId();
    }

}

File: 13_NoSQL/examsdb-java/src/main/java/at/spengergasse/examsdb/model/Exam.java
package at.spengergasse.examsdb.model;

import java.time.ZonedDateTime;

import org.bson.codecs.pojo.annotations.BsonId;
import org.bson.types.ObjectId;

public record Exam(@BsonId ObjectId id, StudentName student, TeacherName teacher,
        SchoolClass currentClass, SchoolClass examClass, Subject subject, ZonedDateTime dateTime,
        int pointsMax, int points, int grade) {
    public Exam {
    }

    public Exam(StudentName student, TeacherName teacher,
            SchoolClass currentClass, SchoolClass examClass, Subject subject, ZonedDateTime dateTime,
            int pointsMax, int points, int grade) {
        this(new ObjectId(), student, teacher, currentClass, examClass, subject, dateTime, pointsMax, points, grade);
    }
}